src/main/kotlin/no/nav/syfo/client/pdl/PdlClient.kt

```diff
@@ -23,82 +22,47 @@ class PdlClient(
     private val redisStore: RedisStore,
     private val httpClient: HttpClient = httpClientDefault(),
 ) {
-    suspend fun getPersonWithOboToken(
+    suspend fun getPerson(
         callId: String,
         personident: Personident,
-        token: Token?,
-    ) = person(
-        callId = callId,
-        personident = personident,
-        token = token,
-    )
-
-    suspend fun getPersonWithSystemToken(
-        callId: String,
-        personident: Personident,
-    ) = person(
-        callId = callId,
-        personident = personident,
-        token = null,
-    )
-
-    private suspend fun person(
-        callId: String,
-        personident: Personident,
-        token: Token?,
-    ): PdlHentPerson {
+    ): PipPersondataResponse {
         val cacheKey = "$PDL_PERSON_CACHE_KEY-$personident"
-        val cachedPerson = getCachedPdlHentPerson(cacheKey)
+        val cachedPerson = getPipPersondataResponse(cacheKey)
 
         return if (cachedPerson != null) {
             cachedPerson
         } else {
-            val pdlHentPerson = getPersonFromPdl(callId, personident, token)
-            redisStore.setObject(key = cacheKey, value = pdlHentPerson, expireSeconds = TWELVE_HOURS_IN_SECS)
-            pdlHentPerson
+            getPersonFromPdl(callId, personident).also {
+                redisStore.setObject(
+                    key = cacheKey,
+                    value = it,
+                    expireSeconds = TWELVE_HOURS_IN_SECS,
+                )
+            }
         }
     }
 
-    private fun getCachedPdlHentPerson(cacheKey: String): PdlHentPerson? {
+    private fun getPipPersondataResponse(cacheKey: String): PipPersondataResponse? {
         return redisStore.getObject(key = cacheKey)
     }
 
-    private suspend fun getPersonFromPdl(callId: String, personident: Personident, token: Token?): PdlHentPerson {
-        val newToken = if (token == null) {
-            azureAdClient.getSystemToken(
-                scopeClientId = clientId,
-                callId = callId,
-            )
-        } else {
-            azureAdClient.getOnBehalfOfToken(
-                scopeClientId = clientId,
-                token = token,
-                callId = callId,
-            )
-        }?.accessToken
+    private suspend fun getPersonFromPdl(callId: String, personident: Personident): PipPersondataResponse {
+        val token = azureAdClient.getSystemToken(
+            scopeClientId = clientId,
+            callId = callId,
+        )?.accessToken
```

Comment:
Spurte vi om de hadde opplegg for noe annet enn systemtoken? Er det problematisk at det er via systemtoken egentlig?

Reply:
For dette API'et er det bare system-token, og den intenderte bruken er nettopp fra app'er som istilgangskontroll:

Se et stykke nedi denne tråden: https://nav-it.slack.com/archives/C84H68ESC/p1699606343474129